src/recommender.py

```diff
@@ -36,36 +39,137 @@ def _process_evt(self, speaker_id, evt):
       return
 
     action = ACTIONS[action_idx]
-    err = self._send_action(action)
+    empathid, err = self._send_action(speaker_id, action)
 
     # if send recommendation successfully
-    if not err:
-      reward = self.get_reward(ctx, action_idx)
-      self.model.update(ctx, action_idx, reward)
+    if not err and empathid:
+      err, reward = self.get_reward(empathid, ctx, action_idx)
+      if not err:
+        self.model.update(ctx, action_idx, reward)
 
-  def get_reward(self, ctx, action_idx):
+  def get_reward(self, empathid, ctx, action_idx):
     '''
     temp mocked reward
     '''
+    # connect to database
+    db = pymysql.connect('localhost', 'root', '', 'ema')
+    cursor = db.cursor()
+    current_time = time.time()
+    err = None
 
-    reward, _ = self.mock_scenario.insight(0, ctx, action_idx)
-
-    return reward
-
-
-  def _send_action(self, action):
+    time_count = 0
+    reward = ""
+
+
+    #determine variablename =
+    if ((action_idx + 1) >= 0) and ((action_idx +1) <=9):
+      var_name_code = "00"+str(action_idx+1)
+    elif ((action_idx + 1) >= 10) and ((action_idx +1) <=99):
+      var_name_code = "0"+str(action_idx+1)
+    elif ((action_idx + 1) >= 100) and ((action_idx +1) <=999):
+      var_name_code = str(action_idx+1)
+
+    # recieving reward from user
+    while time.time() - current_time < 30:
+      # print(str(empathid))
+      query = "SELECT answer FROM ema_data where primkey = '1:" + empathid + "' AND variablename = 'R" + var_name_code + "Q01'"
+      data = cursor.execute(query)
+
+      # if the user took some action
+      if data:
+        # empathid = '999|' + str(int(time.time()))
+        answer = str(cursor.fetchall()).split("'")[1]
+        # print('answer:', answer)
+
+        # if NO return 1
+        if answer == '2':
+          reward = 0.0
+          time_count += 1
+          break
+        # if YES return 0
+        if answer == '1':
+          reward = 1.0
+          time_count += 1
+          break
+
+    if time_count == 0:
+      err = "Timeout Error"
+    print(reward)
```

Comment:
the `print` logging is for development, unnecessary in the deployed system. pls remove it